python_modules/dagster/dagster/core/execution_context.py

```diff
@@ -310,117 +195,179 @@ def log(self):
         pass
 
 
-class WithLegacyContext(six.with_metaclass(ABCMeta)):  # pylint: disable=no-init
-    @abstractproperty
-    def legacy_context(self):
-        pass
+class PipelineContextData(
```

Comment:
for consistency, even if it's slightly more verbose, can we settle either on `PipelineContext` or on `PipelineExecutionContext`, and use it everywhere? the more consistent the easier to navigate